dialer.go

```diff
@@ -226,6 +230,39 @@ func (d *Dialer) connectTLS(ctx context.Context, conn net.Conn) (tlsConn *tls.Co
 	return
 }
 
+func (d *Dialer) authenticateSASL(ctx context.Context, conn net.Conn) error {
+	// Temporary create a new conn.
+	// TODO: Can we move authentication to a later step when Conn is available?  Is it an issue that we might reuse correlation ids?
+	c := NewConn(conn, "", 0)
+
+	c.saslHandshake(saslHandshakeRequestV1{Mechanism: d.SASLClient.Mechanism()})
+
+	// TODO: Create a new SASL client instance per authenticate SASL call,
+	// or make sure we do not use this instance concurrently and document
+	// that Start() can be called multiple times
+	bytes, err := d.SASLClient.Start(ctx)
```

Comment:
re. factory comment above, it could be used like this:

```suggestion
        client := d.SASLClient()
	bytes, err := client.Start(ctx)
```